backend/canvas_app_explorer/views.py

```diff
@@ -1,15 +1,33 @@
-from django.shortcuts import render
-
+from canvas_oauth.oauth import get_oauth_token
+from django.conf import settings
 from rest_framework import permissions, viewsets
+from rest_framework.request import Request
 
 from backend.canvas_app_explorer import models, serializers
-
+from .canvas_lti_manager import CanvasLtiManager
 
 class LTIToolViewSet(viewsets.ModelViewSet):
     """
-    API endpoint that allows users to be viewed or edited.
+    API endpoint that lists LTI tools available in the course context
     """
-    queryset = models.LtiTool.objects.all()
     serializer_class = serializers.LtiToolSerializer
+    permission_classes = [permissions.DjangoModelPermissions]
+    queryset = models.LtiTool.objects.all()
+
+    def get_serializer_context(self):
+        context = super(LTIToolViewSet, self).get_serializer_context()
+        request: Request = context['request']
+        access_token = get_oauth_token(request)
+        manager = CanvasLtiManager(
```

Comment:
We probably could simplify this `CanvasLtiManager` so you only need to call it with 2 parameters most of the time. (Or all of the time)

So you'd be like `CanvasLtiManager(api_key=access_token, course_id=request.session['course_id']`). If could just pull that `api_url` parameter directly.

Actually, we could probably also just change `CanvasLtiManger` to just take a single parameter request. Then it would be even less code outside. I just didn't see how it was going to be used yet.

Reply:
Yeah,  I was thinking that just to simplify all of the code that will be calling into it.

I mean we could make the method take multiple parameters with some default values and figure out what to do based on what was passed too since you can't overload constructors in Python.

I think this is fine for now but I don't really want to copy/paste 3-4 LOC every time we initialize this object. Maybe I'll revise this when I add more methods to the API.

Reply:
I get it, you don't want to repeat things too much. However, I also realized that making the manager class know about requests and environment variables limits our ability to write tests, I think.

If we wanted to make both of us happy, we could create a factory?

Something like this:
```
class DjangoCourseLTIManagerFactory():
    
    def __init__(api_url: string):
        self.api_url = api_url

    createManager(request: Request) -> CanvasLTIManager:
        # Logic setting up access_token and course ID go here
        course_id = request.session['course_id']
        token = get_access_token(request)
        return CanvasLTIManager(self.api_url, token, course_id)
```

And then initialize a factory instance somewhere and share it between modules. WDYT?

Reply:
I'll merge this PR, and we can discuss this idea as part of your work.

Reply:
This is an idea. I think we'd probably want two methods, one that can take a Request, or one that can take course id and token. 

And also api_url could be defaulted `settings.CANVAS_OAUTH_CANVAS_DOMAIN` if not set. 

Fine to work on this independently, and maybe don't even need it now. Just want to make this as easy/brief to call if we'll be calling it a lot.